src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -158,7 +158,7 @@ internal override GeneratedLabelSymbol ContinueLabel
 
         internal override TypeSymbol GetIteratorElementType(YieldStatementSyntax node, DiagnosticBag diagnostics)
         {
-            TypeSymbol returnType = _methodSymbol.ReturnType;
+            TypeSymbol returnType = (_methodSymbol as LocalFunctionSymbol)?.ReturnTypeIterator ?? _methodSymbol.ReturnType;
```

Comment:
No, it's an ErrorType. The whole ReturnTypeIterator and ReturnTypeNoForce are rather hacky things, I tried to solve it elegantly but was unable to.

In this case, ReturnTypeIterator conveys to the return type calculator that "var" should not be allowed. Calling .ReturnType directly results in different behavior - var *is* allowed. And the other case, ReturnTypeNoForce, should be used when inside the method and calculating if a "return" statement matches the return type of the current method - if it's "var", we don't know it yet, so return null. (it is checked later in the inference phase if it's valid)